### PR TITLE
fix: data and ecocredit gas consumption

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #185

This pull request fixes and updates gas consumption for the data module and ecocredit module. Ensures gas consumption for all iterations as well as hash verification in the data module. Also updates gas consumption descriptions.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/data/gas.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package data
+
+// TODO: Revisit this once we have proper gas fee framework.
+// Tracking issue https://github.com/cosmos/cosmos-sdk/discussions/9072
+const GasCostPerIteration = uint64(10)
```
Comment (robert-zaremba) [inline x/data/gas.go:5]: seams we are copying this value all over different places. Can we use it form a single place?
I know we are using a different modules, but maybe we can create some params module for that. (not necessary in this task). Something to discuss, not necessary do in this PR.

**Review by robert-zaremba — APPROVED** [review]
nice improvement.

File: x/data/gas.go (hunk 1 above)
Comment (ryanchristo) [reply x/data/gas.go:5]: I like the idea. Although if we want to set module specific values, we might want to keep them within each module, same way we do module specific errors. Currently we use `10` for all iterations but we might want to have some iterations charge more gas than others. I thought about making those changes here but figured I would start with making sure we are at least charging some gas for each iteration.

File: x/ecocredit/server/core/retire.go [hunk 2]
```diff
@@ -2,6 +2,7 @@ package core
 
 import (
 	"context"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
```
Comment (technicallyty) [inline x/ecocredit/server/core/retire.go:5]: some of these imports didn't appear to join the existing /regen-ledger/ group

**Review by technicallyty — APPROVED** [review]
lgtm - some files have imports ungrouped

File: x/ecocredit/server/core/retire.go (hunk 2 above)
Comment (ryanchristo) [reply x/ecocredit/server/core/retire.go:5]: I just updated my golang preferences too. Thanks for sharing that over at the watercooler. :slightly_smiling_face:

**Pull request merged**